### ISSUE TITLE
keystone build --frozen does not exit when schemas are mismatched

### ISSUE THREAD

**Issue by Thinkscape** [keystonejs/keystone#8504]
### Repro steps
1. Having a keystone project with a `graphql.schema` in root dir
2. Modify the keystone schema
3. Run `keystone build --frozen` manually, or as part of `postinstall` script

### Expected

 As [per the documentation](https://keystonejs.com/docs/guides/cli#build-flags) the script should exit with code `> 0` as the schemas are mismatched

### Actual

 The script waits for input and asks if I want to update the schema:
```
> keystone build --frozen

Your Prisma schema is not up to date
✔ Replace the Prisma schema? … 
```

### Env
```
node v18.15.0
@keystone-6/core@5.1.0
```

**Comment by borisno2** [comment]
Hi @Thinkscape 
Currently the `--frozen` flag will check if `process.env.SKIP_PROMPTS` is set, which is automatically set in CI, and if set will `throw new ExitError(1)` otherwise with prompt to update. This might need to be clearer in the documentation.
 
Also, open to feedback if there is a good use case for this to be different. But the general Idea is this Exits with an error in CI but prompts when it is able.

**Comment by borisno2** [comment]
> @borisno2 are we using `is-ci` for that?

No, we are using `export let shouldPrompt = process.stdout.isTTY && !process.env.SKIP_PROMPTS;` in `pacakges/core/src/lib/prompts.ts`
https://github.com/keystonejs/keystone/blob/818b0b61a1367c39fdc4d2b91f69fe8a4a157adc/packages/core/src/lib/prompts.ts#L31

**Comment by Thinkscape** [comment]
Local dev `postinstall` would not have `SKIP_PROMPTS` present - used for the first build of git-ignored tsgql and prisma client.
